sdks/python/apache_beam/ml/transforms/embeddings/huggingface.py

```diff
@@ -130,6 +142,88 @@ def get_model_handler(self):
   def get_ptransform_for_processing(self, **kwargs) -> beam.PTransform:
     # wrap the model handler in a _TextEmbeddingHandler since
     # the SentenceTransformerEmbeddings works on text input data.
-    return RunInference(
-        model_handler=_TextEmbeddingHandler(self),
-        inference_args=self.inference_args)
+    return (
+        RunInference(
+            model_handler=_TextEmbeddingHandler(self),
+            inference_args=self.inference_args,
+        ))
+
+
+class _InferenceAPIHandler(ModelHandler):
+  def __init__(self, config: 'InferenceAPIEmbeddings'):
+    super().__init__()
+    self._config = config
+
+  def load_model(self):
+    session = requests.Session()
+    session.headers.update(self._config.authorization_token)
+    return session
+
+  def run_inference(
+      self, batch, session: requests.Session, inference_args=None):
+    response = session.post(
+        self._config.api_url,
+        headers=self._config.authorization_token,
+        json={
+            "inputs": batch, "options": inference_args
+        })
+    return response.json()
+
+
+class InferenceAPIEmbeddings(EmbeddingsManager):
+  def __init__(
+      self,
+      hf_token: str,
```

Comment:
Should we make this optional? It looks like this kind of authorization can also be set in the environment - https://huggingface.co/docs/huggingface_hub/package_reference/environment_variables#hftoken - we could try to pull the token using that env variable if this isn't set. The downside is that it makes it a runtime error, but it would probably be a more robust/secure approach if users have custom containers.

Reply:
Done